rust/lance/src/catalog/catalog_trait.rs

```diff
@@ -0,0 +1,276 @@
+            uri: ds.uri().to_string(),
+            object_store: self.object_store.clone(),
+            commit_handler: self.commit_handler.clone(),
+            store_params: Some(ObjectStoreParams {
+                object_store: Some((
+                    self.object_store.clone().inner,
+                    Url::parse(&table_url).unwrap(),
+                )),
+                ..Default::default()
+            }),
+        })
+    }
+
+    async fn drop_table(&self, name: &str) -> Result<()> {
+        let table_path = format_table_url_or_path(
+            false,
+            &self.object_store.scheme,
+            self.base_path.as_ref(),
+            name,
+        );
+
+        self.object_store.remove_dir_all(table_path).await?;
+
+        Ok(())
+    }
+
+    async fn get_table(&self, name: &str) -> Result<TableReference> {
+        let table_url = format_table_url_or_path(
+            true,
+            &self.object_store.scheme,
+            self.base_path.as_ref(),
+            name,
+        );
+
+        let ds = DatasetBuilder::from_uri(&table_url)
+            .with_commit_handler(self.commit_handler.clone())
+            .with_object_store(
+                self.object_store.clone().inner,
+                Url::parse(&table_url).unwrap(),
+                self.commit_handler.clone(),
+            )
+            .load()
+            .await?;
+
+        Ok(TableReference {
+            uri: ds.uri().to_string(),
+            object_store: self.object_store.clone(),
+            commit_handler: self.commit_handler.clone(),
+            store_params: Some(ObjectStoreParams {
+                object_store: Some((
+                    self.object_store.clone().inner,
+                    Url::parse(&table_url).unwrap(),
+                )),
+                ..Default::default()
+            }),
+        })
+    }
+
+    async fn list_tables(
+        &self,
+        start_after: Option<String>,
+        limit: Option<u32>,
+    ) -> Result<Vec<String>> {
+        let mut f = self
+            .object_store
+            .read_dir(self.base_path.clone())
+            .await?
+            .iter()
+            .map(std::path::Path::new)
+            .filter(|path| {
+                let is_lance = path
+                    .extension()
+                    .and_then(|e| e.to_str())
+                    .map(|e| e == LANCE_EXTENSION);
+                is_lance.unwrap_or(false)
+            })
+            .filter_map(|p| p.file_stem().and_then(|s| s.to_str().map(String::from)))
+            .collect::<Vec<String>>();
+        f.sort();
+        if let Some(start_after) = start_after {
+            let index = f
+                .iter()
+                .position(|name| name.as_str() > start_after.as_str())
+                .unwrap_or(f.len());
+            f.drain(0..index);
+        }
+        if let Some(limit) = limit {
+            f.truncate(limit as usize);
+        }
+        Ok(f)
+    }
+}
+
+pub struct ListingCommitHandler {
```

Comment:
What's the purpose of desiging this `ListingCommitHandler` ?  especialy for a `Catalog` ?  The Catalog usually don't touch or write the dataset manifests or transactions for the dataset/table, so therioally we don't need to introduce the `CommitterHandler`, right ?

Reply:
If the Lance format has more complete APIs, you are correct.

For example, it could have a Rust drop dataset API to handle drop data and drop manifest with an atomical operation.

However, currently, it has not, so it needs the commit handler to clear metadata(manifest somethings controls via external commit handler).